OMF_API/Program.cs

```diff
@@ -1,540 +1,362 @@
             {
-                Console.WriteLine(ex.ToString());
-
-                toThrow = ex;
+                Console.WriteLine($"Encountered Error: {e}");
                 success = false;
+                if (test)
+                    throw;
             }
-            finally
-            {
-                Console.WriteLine("Deleting");
-                //step 10
-                try
-                {
-                    deleteTypesAndContainers();
-                }
-                catch (Exception ex)
-                {
-                    Console.WriteLine(ex.ToString());
-
-                    toThrow = ex;
-                    success = false;
-                }
 
-                CheckDeletes();
+            Console.WriteLine("Done");
+            return success;
+        }
 
-                Console.WriteLine("Done");
-                if (!test)
-                    Console.ReadLine();
-                if (toThrow != null)
-                    throw toThrow;
-            }
+        /// <summary>
+        /// Gets a json file in the current directory of name filename.
+        /// </summary>
+        /// <param string="filename">The file to retrieve</param>
+        /// <returns></returns>
+        public static dynamic getJsonFile(string filename)
+        {
+            dynamic dynamicJson = JsonConvert.DeserializeObject(File.ReadAllText($"{Directory.GetCurrentDirectory()}/{filename}"));
 
-            return success;
+            return dynamicJson;
         }
 
         /// <summary>
-        /// Checks to mkae sure things were deleted
+        /// Gets the application settings
         /// </summary>
-        private static void CheckDeletes()
+        /// <returns></returns>
+        public static AppSettings getAppSettings()
         {
-            Console.WriteLine("Check Deletes");
-            Console.WriteLine("Letting OMF get to data store");
-            Thread.Sleep(10000);
-            bool success = false;
-            if (sendingToOCS)
+            AppSettings settings = JsonConvert.DeserializeObject<AppSettings>(File.ReadAllText(Directory.GetCurrentDirectory() + "/appsettings.json"));
+
+            // check for optional/nullable parameters and invalid endpoint types
+            foreach (var endpoint in settings.Endpoints)
             {
-                try
-                {
-                    checkValue(checkBase + $"/Streams" + $"/Container1");
-                }
-                catch
-                {
-                    success = true;
-                }
-                if (!success)
-                    throw new Exception("Container was found.");
+                if (endpoint.VerifySSL == null)
+                    endpoint.VerifySSL = true;
+                if (!(string.Equals(endpoint.EndpointType, "OCS") || string.Equals(endpoint.EndpointType, "EDS") || string.Equals(endpoint.EndpointType, "PI")))
+                    throw new Exception($"Invalid endpoint type {endpoint.EndpointType}");
```

Comment:
you can ignore the previous comment about error handling outside of these cases since it's done here. making this an enum is still something to think about

Reply:
The `string.Equals()` calls here should use the overload with `StringComparisonType.OrdinalIgnoreCase` or some version of `IgnoreCase` though.